autosklearn/automl.py

```diff
@@ -347,7 +347,13 @@ def fit(
                                      'valid feature types, you passed `%s`' % ft)
 
         # Feature types dynamically understood from dataframe
-        if feat_type is None and self.InputValidator.feature_types:
+        if feat_type is not None and self.InputValidator.feature_types:
```

Comment:
Could you please also add a unit test for this?

Reply:
Yes thanks, I will add unit test/support to make sure that the dtype does not change once fitted.